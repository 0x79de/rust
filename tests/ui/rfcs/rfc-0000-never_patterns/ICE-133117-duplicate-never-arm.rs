#![feature(never_patterns)]
#![allow(incomplete_features)]

fn main() {
    match () {
        (!|
        //~^ ERROR: mismatched types
        !) if true => {} //~ ERROR a never pattern is always unreachable
        //~^ ERROR: mismatched types
        (!|!) if true => {} //~ ERROR a never pattern is always unreachable
    }
}
