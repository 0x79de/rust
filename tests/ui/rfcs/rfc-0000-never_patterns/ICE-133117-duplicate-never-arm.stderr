error: a never pattern is always unreachable
  --> $DIR/ICE-133117-duplicate-never-arm.rs:8:23
   |
LL |         !) if true => {}
   |                       ^^
   |                       |
   |                       this will never be executed
   |                       help: remove this expression

error: a never pattern is always unreachable
  --> $DIR/ICE-133117-duplicate-never-arm.rs:10:26
   |
LL |         (!|!) if true => {}
   |                          ^^
   |                          |
   |                          this will never be executed
   |                          help: remove this expression

error: mismatched types
  --> $DIR/ICE-133117-duplicate-never-arm.rs:6:10
   |
LL |         (!|
   |          ^ a never pattern must be used on an uninhabited type
   |
   = note: the matched value is of type `()`

error: mismatched types
  --> $DIR/ICE-133117-duplicate-never-arm.rs:8:9
   |
LL |         !) if true => {}
   |         ^ a never pattern must be used on an uninhabited type
   |
   = note: the matched value is of type `()`

error: aborting due to 4 previous errors

